### PR TITLE
Render full dashboard and single URL dashboard via Widgets API

### PR DESCRIPTION
**Pull request by felixarntz**
## Summary

Addresses issue #2646

## Relevant technical choices

* I diverged from the IB a little bit in that I introduced a general `PageFooter` component which is the counterpart of `PageHeader`. I still went with individual wrappers specific to `Dashboard*` and `DashboardDetails*` to have these decoupled, but I think overall it makes sense to have a central point to control the overall page footer. I also changed everywhere that same type of help link is displayed to use the new `PageFooter` component.
* I removed some unnecessary legacy wrapper components, such as `DashboardMain`, `DashboardModules`, and `DashboardDetailsModules`. Only the filtered component remains (renamed including `Legacy*` prefix).
* I also removed `DashboardDetailsEntityHeaderContainer` since that was now only being used in one place and it was just a layout grid, so in no way specific to this use-case.

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Review by aaemnnosttv — COMMENTED** [review]
LGTM, just missing one file header 😄 

There was an extra `key` prop in a few places that doesn't seem to be necessary either but not really a blocker. See my comments below.

**Review by aaemnnosttv — APPROVED** [review]
Nice work!

File: assets/js/components/dashboard/DashboardApp.js [hunk 1]
```diff
@@ -45,30 +45,32 @@ export default function DashboardApp() {
...
 						<Row>
 							<Cell size={ 12 }>
-								<PageHeader
-									className=" googlesitekit-heading-2 googlesitekit-dashboard__heading"
-									title={ __( 'Site Overview', 'google-site-kit' ) }
-								/>
+								<DashboardHeader />
+							</Cell>
+							<LegacyDashboardModule
+								key={ 'googlesitekit-dashboard-module' }
```
[the remark this answers is not in the document]
Comment (felixarntz) [reply assets/js/components/dashboard/DashboardApp.js:65]: Yeah I left it in from before. Shouldn't hurt :)

File: assets/js/components/dashboard-details/DashboardDetailsApp.js [hunk 2]
```diff
@@ -49,44 +50,35 @@ export default function DashboardDetailsApp() {
...
+			{ ! pageDashboardWidgetsEnabled && (
+				<div className="googlesitekit-module-page googlesitekit-dashboard-single-url">
+					<Grid>
+						<Row>
+							<Cell size={ 12 }>
+								<DashboardDetailsHeader />
+							</Cell>
+							{ currentEntityURL && (
+								<LegacyDashboardDetailsModule
+									key={ 'googlesitekit-dashboard-details-module' }
```
[the remark this answers is not in the document]
Comment (felixarntz) [reply assets/js/components/dashboard-details/DashboardDetailsApp.js:72]: Same as above.

**Pull request merged**